docs/testing/workspaces-migration-guide.md

```diff
@@ -0,0 +1,319 @@
+
+1. Create a `CallBuilder` by invoking `Contract::call`
+2. Pass function call arguments via `CallBuilder::args_json` or `CallBuilder::args_borsh` depending on which serialization algorithm your contract is using
+3. Configure gas and deposit (if needed) via `CallBuilder::gas` and `CallBuilder::deposit`
+4. Finalize the call by consuming builder via `CallBuilder::transaction` or `CallBuilder::view` depending on what kind of call you want to make
+
+Reference this migration of Example 1 for migrating your own calls:
+
+```rust title="Calls - workspaces-rs"
+contract
+    .call(&worker, "ft_transfer")
+    .args_json((alice.id(), transfer_amount, Option::<bool>::None))?
+    .gas(300_000_000_000_000)
+    .deposit(ONE_YOCTO)
+    .transact()
+    .await?;
+
+let root_balance: U128 = contract
+    .call(&worker, "ft_balance_of")
+    .args_json((contract.id(),))?
+    .view()
+    .await?
+    .json()?;
+```
+
+:::note
+Note that you have to pass arguments as any serializable type representing a sequential list. Tuples are usually the best candidate due to their heterogeneous nature (remember that you can construct a unary tuple by placing a comma before the closing bracket like this: `(el,)`).
+:::
+
+### Batch transactions
+
+There is a special builder for making batch transactions that can be instantiated by calling `Contract::batch`. Consider the following snippet making a batch transaction consisting of two calls:
+
+```rust title="Batch Transaction - near-sdk-sim"
+let res = root
+    .create_transaction(contract.account_id())
+    .function_call(
+        "ft_transfer_call".to_string(),
+        json!({
+            "receiver_id": defi_contract.account_id(),
+            "amount": transfer_amount.to_string(),
+            "msg": "10",
+        })
+        .to_string()
+        .into_bytes(),
+        300_000_000_000_000 / 2,
+        1,
+    )
+    .function_call(
+        "storage_unregister".to_string(),
+        json!({
+            "force": true
+        })
+        .to_string()
+        .into_bytes(),
+        300_000_000_000_000 / 2,
+        1,
+    )
+    .submit();
+```
+
+There are no caveats here, the snippet can be straightforwardly mapped into the following:
+
+```rust title="Batch Transaction - workspace-rs"
+let res = contract
+    .batch(&worker)
+    .call(
+        Function::new("ft_transfer_call")
+            .args_json((defi_contract.id(), transfer_amount, Option::<String>::None, "10"))?
+            .gas(300_000_000_000_000 / 2)
+            .deposit(1),
+    )
+    .call(
+        Function::new("storage_unregister")
+            .args_json((Some(true),))?
+            .gas(300_000_000_000_000 / 2)
+            .deposit(1),
+    )
+    .transact()
+    .await?;
+```
+
+## Inspecting logs
+
+:::danger
+TODO: I found some potential inconsistencies in what I wrote about the log inspection behavior, need some time to look at this further
```

Comment:
what kind of inconsistencies did you find. The section you wrote afterwards describes it well